Libplanet/Net/Transports/NetMQTransport.cs

```diff
@@ -914,27 +893,31 @@ private async Task ProcessRequest(MessageRequest req, CancellationToken cancella
                 try
                 {
                     await Task.Delay(period, cancellationToken);
-                    ImmutableHashSet<Address> peerAddresses =
-                        _table.Peers.Select(p => p.Address).ToImmutableHashSet();
                     foreach (Address address in _dealers.Keys)
                     {
-                        if (!peerAddresses.Contains(address) &&
-                            _dealers.TryRemove(address, out DealerSocket removed))
+                        if (DateTimeOffset.UtcNow - _dealers[address].Item2 > _dealerSocketLifetime
+                            && _dealers.TryRemove(
+                                address,
+                                out (DealerSocket DealerSocket, DateTimeOffset) pair))
                         {
-                            removed.Dispose();
+                            pair.DealerSocket.Dispose();
                         }
```

Comment:
Although chances are slim, checking an item *and* removing an item are two separate operations, hence prone to a race condition, i.e. not thread safe. For instance, `_dealers[address]` can be determined to be stale and right before `_dealers.TryRemove()` is run, another thread can try to use `_dealers[address]` socket which will be disposed immediately by the thread controlled by `DisposeUnusedDealerSockets()`. The easiest solution would be to lock `_dealers`.

Reply:
Added in 7ba2aab189a551166a66601ad16516348c947055. PTAL :)